pkg/controller/controller.go

```diff
@@ -406,15 +414,21 @@ func (c *Controller) handleActionSet(as *crv1alpha1.ActionSet) (err error) {
 	return nil
 }
 
+func (c *Controller) LoadOrStoreTomb(ctx context.Context, asName string) (*tomb.Tomb, context.Context) {
+	var t *tomb.Tomb
+	if v, ok := c.actionSetTombMap.Load(asName); ok {
+		t = v.(*tomb.Tomb)
+	} else {
+		t, ctx = tomb.WithContext(ctx)
+		c.actionSetTombMap.Store(asName, t)
+	}
+	return t, ctx
+}
```

Comment:
nit (to avoid using `else`):

```suggestion
	var t *tomb.Tomb
	if v, ok := c.actionSetTombMap.Load(asName); ok {
		t = v.(*tomb.Tomb)
		return t, ctx
	t, ctx = tomb.WithContext(ctx)
	c.actionSetTombMap.Store(asName, t)
	return t, ctx
}
```

Reply:
Fixed